### PR TITLE
⬆️(backend) upgrade to python 3.11

### PR DESCRIPTION
**Pull request by jonathanreveille**
The application was running in python 3.10, it was about time to upgrade it into python 3.11. 

In this upgrade, we have upgraded  the cookiecutter folder with the version of python to 3.11.

## Purpose

Upgrading the application to python 3.11
Make bootstrap works. When using the slim version of `python:3.11-slim-bookworm`, the command who failed saying that there is an issue with the package :  `setuptools-scm>=8.0`. I've tried to take the heavier version, and make bootstrap works like a charm.

## Proposal
- [x] Update Circle CI to use python 3.11 image for main project.
- [x] Update pyproject.toml classifiers to mention the possible use of python 3.11.
- [x] Update cookiecutter section for its DockerFile and CircleCi .yml configuration files.

Must wait for this [PR](https://github.com/openfun/richie/pull/2399) to be merged before merging this one.

Version of Docker image for python 3.11 : 
- **python:3.11-bookworm** :  `make bootstrap`  works (will use this one because 3.11.9 has the same hash with 3.11)
- **python:3.11.9-bookworm** :  `make bootstrap` works 
- **python:3.11-slim-bookworm** : fails when `make bootstrap` with this package `setuptools-scm>=8.0`.
- **python:3.11.9-slim-bookworm** : fails when `make bootstrap` with this package  `setuptools-scm>=8.0`.
- **python:3.11.9-bullseye** : `make bootstrap` works

### CODE REVIEW

File: tests/apps/search/test_indexers_courses.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def test_indexers_courses_get_es_document_no_image_icon_picture(self):
             course_icons_placeholder,
             CategoryPlugin,
             "en",
-            **{"page": category.extended_object}
+            **{"page": category.extended_object},
         )
         course.extended_object.publish("en")
         # Make sure we associate an image-less picture with the category through
```
Comment (jbpenrath) [inline tests/apps/search/test_indexers_courses.py:591]: You should revert this change.

**Review by jbpenrath — APPROVED** [review]
I do not understand the issue with bookworm ?

**Review by lunika — APPROVED** [review]

**Review by kernicPanel — APPROVED** [review]

File: tests/apps/search/test_indexers_courses.py (hunk 1 above)
Comment (jonathanreveille) [reply tests/apps/search/test_indexers_courses.py:591]: The command `make lint-back` adds this correction of a missing comma.

**Review by lunika — APPROVED** [review]
